### DISCUSSION TITLE
Juan Campa - membrane.io

### DISCUSSION THREAD

**Discussion by just-be-dev** [devtools-fm/devtools.fm#126] [Announcements]
## Background

- [Tweet](https://twitter.com/juancampa/status/1591617650870091776) that informed us of the project
- [membrane.io](https://www.membrane.io/)
- https://twitter.com/juancampa
- [membrane demo](https://www.youtube.com/watch?v=nwf-yVVJJxI&ab_channel=JuanC)

## Tools in this space
- https://interval.com/
- https://autocode.com/
- https://www.val.town/
- https://n8n.io/

## Themes
- Simplifying data persistence in programs
- Enhancing programming environments with browser extensions
- Visualizing APIs on sites

## Intro
**Andrew**: Hello, welcome to the devtools.fm podcast. This is a podcast about developer tools and the people who make them. I'm Andrew and this is my co-host Justin.

**Justin**: Hey everyone! Our guest today is Juan Campa. Juan is the creator of membrane.io, a still-in-development platform for simplifying API automation and internal tooling. I'm really excited to chat about membrane, but before we dig into that, Juan, could you tell our listeners a little more about yourself?

### membrane

- [x] Describe membrane for us a bit. What is it and what are you hoping to accomplish?w
- [x] What does the workflow without membrane look like? What other tools might a dev reach for?
- [x] Membrane was originally developed for games. Is that still the case?
- [x] What types of programs do you envision folks building with membrane?
- [x] One of the defining features of membrane is transparent program persistence. You just write code and the state of your app is persisted. Can you talk a little bit about how that works? What trade-offs are you making?
- [x] Does it handle credentials?
- [x] You say it can replace tools like cron. Is it as simple as using a set interval? what actually going on in the background if I set a yearly interval like you birthday text automation?
- [x] You shared a really cool demo on twitter where you visually overlaid what apis were powering data on a GitHub page and then dragged one of those apis into a membrane program. What inspired that? How does it work?
- [x] You've created a browser extension that connects to membrane apps to let you interact with a page directly. Tell us more about that. 
- [x] Membrane works in browsers but what if I want to use to to capture API calls from a mobile device? (or really any type of device other than a desktop browser)

### Future

- [x] What's next for membrane?
- [x] What do you think the future is in this space?

**Comment by just-be-dev** [comment]
Tooltips
- https://www.val.town/
- http://worrydream.com/refs/

**Comment by juancampa** [comment]
Tools I've recently discovered and continually use:

 - [Loop2Learn](https://www.loop2learn.com/): This app accelerates my drums learning by 10X. When practicing a song, you can loop a specific clip (from say, Youtube) and slow it down as much as you need.
 - [Tailscale](https://tailscale.com/): I recently discovered this, and it's fantastic for "hybrid" development: running one service locally while the rest still runs in Kubernetes.

**Comment by hipstersmoothie** [comment]
- https://slashbase.com/#features
- https://turbo.build/pack
